### PR TITLE
feat: Update CoreBiome and SolidRasterizer to use new Biome API for querying block types

### PR DESCRIPTION
**Pull request by naalit**
Now that https://github.com/Terasology/BiomesAPI/pull/12 is merged, the new API can be used for `CoreBiome`s. This doesn't change the generated terrain at all, just moves per-biome code from `SolidRasterizer` to `CoreBiome`.

I'm not sure how the dependency should be changed - should I change the minimum version of BiomesAPI to 4.1.1-SNAPSHOT? The version number hasn't changed since the API change, so it's possible to have BiomesAPI with that version number but without the new API, but maybe that's fine since it's a snapshot version.

### CODE REVIEW

**Comment by keturn** [comment]
Ideally, we'd release BiomesAPI 4.1.1 — or maybe 4.2.0, since it introduces a new API — and then update the dependency of CoreWorlds here to that version.

In practice, everyone is building everything from git all the time anyway, and we haven't keeping track of API stability that closely.

**Review by skaldarnar — DISMISSED** [review]

**Comment by keturn** [comment]
@skaldarnar, there was a dismissed-review from you, it looks like probably from the "update branch" action? Was that an accidentally-dismissed approval?

**Review by skaldarnar — APPROVED** [review]
That was by accident, didn't see the merge conflict before approving, and forgot to re-approve afterwards...

Sometimes "update branch" does not dismiss a review, even if there were merge conflicts, and this one seemed simple enough...

**Pull request merged**
